BungeeCord-Patches/0063-Add-command-to-view-all-proxy-plugins.patch

```diff
@@ -0,0 +1,69 @@
+From becd7e8441c4b43f1b80f18d4ed721321b626b1b Mon Sep 17 00:00:00 2001
+From: MrFishCakes <fishcake007@outlook.com>
+Date: Thu, 20 May 2021 17:43:55 +0100
+Subject: [PATCH] Add command to view all proxy plugins
+
+
+diff --git a/proxy/src/main/java/net/md_5/bungee/BungeeCord.java b/proxy/src/main/java/net/md_5/bungee/BungeeCord.java
+index 00906d3f..5e9fde6d 100644
+--- a/proxy/src/main/java/net/md_5/bungee/BungeeCord.java
++++ b/proxy/src/main/java/net/md_5/bungee/BungeeCord.java
+@@ -75,7 +75,13 @@ import net.md_5.bungee.chat.ScoreComponentSerializer;
+ import net.md_5.bungee.chat.SelectorComponentSerializer;
+ import net.md_5.bungee.chat.TextComponentSerializer;
+ import net.md_5.bungee.chat.TranslatableComponentSerializer;
+-import net.md_5.bungee.command.*;
++import net.md_5.bungee.command.CommandBungee;
++import net.md_5.bungee.command.CommandEnd;
++import net.md_5.bungee.command.CommandIP;
++import net.md_5.bungee.command.CommandPerms;
++import net.md_5.bungee.command.CommandPlugins;
++import net.md_5.bungee.command.CommandReload;
++import net.md_5.bungee.command.ConsoleCommandSender;
+ import net.md_5.bungee.compress.CompressFactory;
+ import net.md_5.bungee.conf.Configuration;
+ import net.md_5.bungee.conf.YamlConfig;
+diff --git a/proxy/src/main/java/net/md_5/bungee/command/CommandPlugins.java b/proxy/src/main/java/net/md_5/bungee/command/CommandPlugins.java
+index 0a8cd585..442c0f3f 100644
+--- a/proxy/src/main/java/net/md_5/bungee/command/CommandPlugins.java
++++ b/proxy/src/main/java/net/md_5/bungee/command/CommandPlugins.java
+@@ -3,6 +3,7 @@ package net.md_5.bungee.command;
+ import net.md_5.bungee.BungeeCord;
+ import net.md_5.bungee.api.ChatColor;
+ import net.md_5.bungee.api.CommandSender;
++import net.md_5.bungee.api.chat.TextComponent;
+ import net.md_5.bungee.api.plugin.Command;
+ import net.md_5.bungee.api.plugin.Plugin;
+ 
+@@ -20,22 +21,23 @@ public class CommandPlugins extends Command {
+         sender.sendMessage(getPluginList());
+     }
+ 
+-    private String getPluginList() {
++    // TODO Implement as Adventure components when added
++    private TextComponent getPluginList() {
+         final Collection<Plugin> plugins = BungeeCord.getInstance().getPluginManager().getPlugins();
+-        final StringBuilder builder = new StringBuilder("Proxy Plugins (" + plugins.size() + "): ");
++        final TextComponent component = new TextComponent("Plugins (" + plugins.size() + "): ");
+ 
+         boolean firstIteration = true;
+         for (Plugin plugin : plugins) {
+             if (!firstIteration) {
+-                builder.append(ChatColor.WHITE).append(", ");
++                component.addExtra(ChatColor.WHITE + ", ");
```

Comment:
This is incorrect usage of BungeeCord's Chat API.
Please actually use the ComponentBuilder.